paasta_tools/cli/cmds/local_run.py

```diff
@@ -653,6 +678,89 @@ def check_if_port_free(port):
     return True
 
 
+def assume_aws_role(
+    instance_config, service, assume_role, assume_pod_identity, aws_profile
+):
+    """Runs AWS cli to assume into the correct role, then extract and return the ENV variables from that session"""
+    pod_identity = instance_config.get_iam_role()
+    if assume_role:
+        pod_identity = assume_role
+    if assume_pod_identity and not pod_identity:
+        print(
+            "Error: --assume-pod-identity passed but no pod identity was found for this instance",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+    default_profile = ""
+    if not aws_profile:
+        with open("/nail/etc/ecosystem") as ecosystem_file:
+            default_profile = ecosystem_file.read()
+    profile = aws_profile or default_profile
```

Comment:
@qlo I think using the current ecosystem and letting the user choose a profile is likely to cause problems, since people might have different profiles set up in depending on their setups (or they may be running this from adhoc-prod, where they don't even have a `~/.aws/credentials` file set up. 

Mentioned in another comment, what do you think about using aws-okta, and choosing an account based on a static mapping from the clustername? e.g. for `-c pnw-prod` you would use `aws-okta -a prod`, for `-c norcal-devc`, you'd use `aws-okta -a dev` 

Then the user can be prompted to choose which role _they_ have access to assume, and then we can use those creds downstream to try and assume the pod identity role

```
$ aws-okta -a prod
Accessing aa-arbitrator socket to fetch Okta cookie (ctrl-c to interrupt) ...
Please choose the role you would like to assume:
[0]: arn:aws:iam::056083216413:role/security
[1]: arn:aws:iam::056083216413:role/security-powerusers
```

I guess the messaging is a little misleading, since it asks you "which role do you want to assume", and then we're going to do some implicit downstream role assumption that they may or may not have access to.

Reply:
This does seem like a better experience. Probably 99.99% of aws profiles here are going to be aws-okta anyway, which is probably the only argument for actually going through AWS cli itself.